pandera/api/pyspark/container.py

```diff
@@ -563,6 +564,34 @@ def to_json(
 
         return pandera.io.to_json(self, target, **kwargs)
 
+    def to_structtype(self) -> StructType:
+        """Recover fields of DataFrameSchema as a Pyspark StructType object.
+
+        As the output of this method will be used to specify a read schema in Pyspark
+            (avoiding automatic schema inference), the False `nullable` properties are
+            just ignored, as this check will be executed by the Pandera validations
+            after a dataset is read.
+
+        :returns: StructType object with current schema fields.
+        """
+        fields = [
+            StructField(column, self.columns[column]._dtype.type, True)
```

Comment:
Would this be able to handle a complex/nested struct method ?

Reply:
Yes, it does and I have just [added a test for it](https://github.com/unionai-oss/pandera/pull/1570/commits/2b56716126548d1d4cec69b81da2206a1c244591).
When we get the annotation from the schema/model, the entire set is copied to the `StructType` object output.

Reply:
sorry I missed this notification. I will take a look at this today.

Reply:
Very minor comment 🙂 

@filipeo2-mck I think it's better to access the public attribute `dtype`, since `_dtype` is a protected attribute. 

```suggestion
            StructField(column, self.columns[column].dtype.type, True)
```

Reply:
Tks for noting that, I just changed it :)